### ISSUE TITLE
Internal error in Agda.TypeChecking.Forcing

### ISSUE THREAD

**Issue by nad** [agda/agda#3063]
The following code, which is accepted by Agda 2.5.3, triggers an internal error:
```agda
postulate
  I  : Set
  U  : I → Set
  El : ∀ {i} → U i → Set

infixr 4 _,_

record Σ (A : Set) (B : A → Set) : Set where
  constructor _,_
  field
    proj₁ : A
    proj₂ : B proj₁

open Σ

∃ : ∀ {A : Set} → (A → Set) → Set
∃ = Σ _

mutual

  infixl 5 _▻_

  data Ctxt : Set where
    _▻_ : (Γ : Ctxt) → Type Γ → Ctxt

  Type : Ctxt → Set
  Type Γ = ∃ λ i → Env Γ → U i

  Env : Ctxt → Set
  Env (Γ ▻ σ) = Σ (Env Γ) λ γ → El (proj₂ σ γ)

mutual

  data Ctxt⁺ (Γ : Ctxt) : Set where
    _▻_ : (Γ⁺ : Ctxt⁺ Γ) → Type (Γ ++ Γ⁺) → Ctxt⁺ Γ

  infixl 5 _++_

  _++_ : (Γ : Ctxt) → Ctxt⁺ Γ → Ctxt
  Γ ++ (Γ⁺ ▻ σ) = (Γ ++ Γ⁺) ▻ σ

data P : (Γ : Ctxt) → Type Γ → Set where
  c : ∀ {Γ σ τ} → P (Γ ▻ σ) τ

f : ∀ {Γ} → Ctxt⁺ Γ → Set₁
f {Γ} (Γ⁺ ▻ σ) = Set
  where
  g : ∀ τ → P (Γ ++ Γ⁺ ▻ σ) τ → Set₁
  g ._ c = Set
```
Error message:
```
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/TypeChecking/Forcing.hs:222
```

**Comment by nad** [comment]
Bisection points to d9359139ab717397279af5719541ebb59c836702: "[ #2888 ] prefer EtaExpandVar over EtaExpandEquation in unification".

**Comment by UlfNorell** [comment]
Our fix of #2888 was too naive. I pushed a better fix, @jespercockx can you have a quick glance to make sure it doesn't do something silly (test suite passes though)?

**Comment by jespercockx** [comment]
Since you only changed the unification strategy there's not much that can go wrong from the point of unification; any strategy that leads to a solution is fine. However, it seems that the problem could be fixed in a simpler way by simply adding the 'EtaExpandVar' suggestion before any occurrence of 'EtaExpandEquation', that way we don't have to guess when 'EtaExpandEquation' will trigger.
